sklearn/linear_model/ridge.py

```diff
@@ -301,8 +312,13 @@ def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
         if np.atleast_1d(sample_weight).ndim > 1:
             raise ValueError("Sample weights must be 1D array or scalar")
 
-        # Sample weight can be implemented via a simple rescaling.
-        X, y = _rescale_data(X, y, sample_weight)
+        if solver != 'sag':
+            # Sample weight can be implemented via a simple rescaling.
+            # FIX: In the sparse case, with fit_intercept=True, we cannot
+            # center the data, so a rescaling is incorrect! See #4710.
+            # Sag solver can actually handle sample weights without rescaling,
+            # and can be used as a reference to solve #4710.
```

Comment:
Could you write something more concise like: "SAG supports sample_weight directly. For other solvers, we implement sample_weight via a simple rescaling."

`fit_intercept=True` is wrong in the sparse case whether we use sample_weight or not. So I don't think the comment above is necessary.

For a temporary fix for #4710, maybe you can set SAG when solver="auto" and fit_intercept=True. Otherwise, raise an exception.